third_party/pyth/pyth_publisher.py

```diff
@@ -147,18 +147,22 @@ def add_symbol(num: int):
 last_new_sym_added_at = time.monotonic()
 
 with ThreadPoolExecutor() as executor: # Used for async adding of products and prices
+    dynamicly_added_symbols = 0
     while True:
         for sym in HTTP_ENDPOINT_DATA["symbols"]:
             publisher_random_update(sym["price"])
 
-        # Add a symbol if new symbol interval configured
-        if PYTH_NEW_SYMBOL_INTERVAL_SECS > 0:
+        # Add a symbol if new symbol interval configured. This will add a new symbol if PYTH_NEW_SYMBOL_INTERVAL_SECS
+        # is passed since adding the previous symbol. The second constraint ensures that
+        # at most PYTH_TEST_SYMBOL_COUNT new price symbols are created.
+        if PYTH_NEW_SYMBOL_INTERVAL_SECS > 0 and dynamicly_added_symbols > PYTH_TEST_SYMBOL_COUNT:
```

Comment:
```suggestion
        if PYTH_NEW_SYMBOL_INTERVAL_SECS > 0 and dynamically_added_symbols > PYTH_TEST_SYMBOL_COUNT:
```

Reply:
I think this should add if it hasn't added enough, instead of acting when there's more than enough.

Reply:
yes you are right.